cmake/OpenCVModule.cmake

```diff
@@ -573,6 +575,10 @@ macro(ocv_create_module)
     "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/cvconfig.h" "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/opencv2/opencv_modules.hpp"
     ${${the_module}_pch})
 
+  set(OPENCV_MODULE_${the_module}_INTERFACE_DEFS ${OPENCV_MODULE_${the_module}_INTERFACE_DEFS}
+    CACHE INTERNAL "Public compile definitions of ${full_modname} module")
+  target_compile_definitions("${the_module}" PUBLIC ${OPENCV_MODULE_${the_module}_INTERFACE_DEFS})
```

Comment:
it would be useful for our internal stuff, but the problem seemed even deeper and it can not be solved. So, I'm closing this MR since it's not needed anymore :(